lms/djangoapps/instructor/enrollment.py

```diff
@@ -361,6 +361,9 @@ def _reset_module_attempts(studentmodule):
     problem_state = json.loads(studentmodule.state)
     # old_number_of_attempts = problem_state["attempts"]
     problem_state["attempts"] = 0
+    problem_state["score_history"] = []
+    problem_state["correct_map_history"] = []
+    problem_state["student_answers_history"] = []
```

Comment:
So this means that the `score_history`, which saves students' scores for calculations (like average, highest, ...), depends on the number of attempts. For example:
- A problem has 3 attempts
- The student submitted their 3 attempts
- The average score is 0.5
- The instructor resets the student's attempt to 0
- The student's score is still 0.5. This is how the platform works; scores are not reset when attempts are.
- The student's score history is empty
- The student tries submitting another answer, it's their first attempt after the reset. The answer is incorrect.
- The student's average score is 0
And so on. Although it makes sense, the product proposal doesn't specify what should happen after resetting attempts. It only mentions this:

> This feature will require to store as part of the problem state the series of scores obtain by the learner in all its attemps, so that the average can always be computed and the grading strategy can be changed at some point in the future and rescore the problem correctly. 

So the premise of this implementation is: 
_All the attempts mean the current attempts, not all time submissions._ 

Is this the correct behavior?

Reply:
I think so.  it would be the correct behavior because when a instructor resets the learner attempts, they are giving the learner the option (not the obligation) of trying the problem again from the starts, even if this creates an inconsistency between the learner grade and the reported number of attemps.   
With the new grading methods, the instructor will also be giving a learner the option, not the obligation to get a new score based on a fresh new set of attempts.
Also, trying to include the submissions prior to the reset in the score calculation would make things very confusing.

Reply:
I agree! It isn't very clear for the learner to include all time submissions. Thanks!